### DISCUSSION TITLE
How to get started + [Feature Request] Allow to take control of the SessionHandler webdriver

### DISCUSSION THREAD

**Discussion by STRTSNM** [jeliebig/WAWebSessionHandler#8] [Ideas]
Hey, I am creating a script that automates WhatsApp. I don't have any idea how to use this module. Can you please share a basic code that demonstrates the method to use it? Thank you

**Comment by TSRBerry** [comment]
Hello,
My script is really easy to use. Just place my script in the directory of your script and import it like this: `from WaWebSession import SessionHandler`

And if you need more guidance on how to use it, I'd like you to read this first: https://github.com/jeliebig/WaWebSessionHandler/blob/master/WaWebSession.py#L461-L481
I believe that this little piece of code can already teach you a lot about the usage of this module, especially if you combine it with the class structure that you can find in README.md (although it is a little outdated atm).

If you need further assistance after that, I'll be happy to help you!

**Reply by STRTSNM** [reply]
Thank you for replaying quick. I learnt how to use your script, but how do I launch a WhatsApp instance in an already opened selenium browser?
My code is :

```
from selenium import webdriver
from selenium.webdriver.chrome.options import Options
from selenium import webdriver;
import requests
from selenium.webdriver.support import expected_conditions as when
from selenium.webdriver.common.by import By as by
from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.firefox.service import Service
from selenium.webdriver.common.keys import Keys
import WaWebSessionHandler as wa_se_h


chrome_options = Options()
chrome_options.add_argument("--user-data-dir=C://Users//home//AppData//Local//Google//Chrome//User Data//Profile 2");
chrome_options.add_argument('--profile-directory=Profile 2')
driver = webdriver.Chrome(chrome_options=chrome_options)
driver.get("https://www.google.com")

handler = wa_se_h.SessionHandler()
handler.access_by_file("sokkaa.cookie")  ## This is a very simple script. Don't judge me
```
I want to open the WhatsApp webpage on the selenium browser that is in line 17. Thank you

**Reply by TSRBerry** [reply]
I see! Actually that was a feature before, but I deleted it as I saw no use for it. I will add another method for you to do that - just give me a few minutes.
I imagine that you would have to adjust your code like this:
`
handler = wa_se_h.SessionHandler(driver=driver)
`
And the line after that should be fine, after I pushed the adjustments. I plan on opening a new tab if the current tab is not blank at the moment, otherwise I load the page in the current tab, so it does not kill anything that you might currently be working on.

I will write another comment as soon as I am done. Thank you for making me aware of this missing feature!

**Reply by TSRBerry** [reply]
Hey again @STRTSNM ,

I've pushed my changes to this repo (e6bb2213902cbe184274946780f38f1c0ec8db7f + 44693de90e16d7853a3041296bef027b59b3f484) and everything should be working like I described before. Make sure to pull the changes or get the file from the repo again. 

Please tell me in case something doesn't work as expected.

**Reply by STRTSNM** [reply]
@jeliebig , Thank you. I will try the updated script and get back to you ASAP.

**Reply by STRTSNM** [reply]
Hello again. The new script is working like a charm, now I am able to open a WhatsApp web instance in an already open browser. but I have a doubt, I get an error: `AttributeError: 'SwitchTo' object has no attribute 'new_window'` . In a line, you have given `if self.__driver.current_url != 'chrome://new-tab-page/' .......`, in that line, I think it means if the current URL is not `chrome://new-tab-page/`. when I change it gives me that error. Any suggestions? Anyways thank you for including this amazing feature 👍

**Reply by TSRBerry** [reply]
Oh that's weird. That should be no problem according to the docs.
Anyways I will try another solution and update the script again. Sorry for the inconvenience and thanks for notifying me about this issue!

**Reply by TSRBerry** [reply]
Done! (https://github.com/jeliebig/WaWebSessionHandler/commit/b570af91f3c5dbe49f719b876ccaaaae132fe04c)
The function I tried to use doesn't exsist outside of the docs, so I couldn't open a new tab and switch to it as easy as I would have liked. But my solution should be fine and only takes 2 additional lines, so I can't really complain about anything.

Hope everything works fine now!

**Reply by STRTSNM** [reply]
Thank you @jeliebig. Now the script works perfectly. Thank you for your continuous support. 👍👍👍👍👍
